src/display/canvas.js

```diff
@@ -3140,8 +3087,8 @@ class CanvasGraphics {
 
       const tmpCanvas = this.cachedCanvases.getCanvas("inlineImage", w, h);
       const tmpCtx = tmpCanvas.context;
-      putBinaryImageData(tmpCtx, imgData, this.current.transferMaps);
-      imgToPaint = tmpCanvas.canvas;
+      putBinaryImageData(tmpCtx, imgData);
+      imgToPaint = this.applyTransferMapsToCanvas(tmpCanvas.canvas);
```

Comment:
Normally you could save a canvas creation in drawing it onto itself (I did that here: https://github.com/mozilla/pdf.js/pull/16075/files#diff-c48e3561004f5db8f11d5ebab2fd661591222ba911cb4173fbced15f026bac6bR1050 and it was working correctly).

Reply:
That's interesting and it'll obviously be more efficient, I didn't know about that; thanks for the pointer!